src/opencl_common.c

```diff
@@ -437,8 +443,9 @@ static void start_opencl_environment()
 {
 	cl_platform_id platform_list[MAX_PLATFORMS];
 	char opencl_data[LOG_SIZE];
-	cl_uint num_platforms, device_num, device_pos = 0;
+	cl_uint num_platforms = 0, device_num = 0, device_pos = 0;
```

Comment:
BTW: `device_num` initialization should be inside the `for`, since platform[n] can be bugged (n > 0).

Reply:
As far as I know, if `clGetPlatformIDs` et. al. returns an error, it didn't touch the passed variable. But you are right, we can rewrite it better. I'll have a look. Drivers are known not to be very trustable anyways...

> BTW: `device_num` initialization should be inside the `for`, since platform[n] can be bugged (n > 0).

That's definitely a valid point, fixed now.

Reply:
Better code in 18186a31c, thanks!